docs/blog/things-to-know-when-developing-with-lit-element/index.md

```diff
@@ -0,0 +1,233 @@
+
+```js
+render() {
+  const closeDelay = this.closeDelay ?? 300;
+  ...
+}
+```
+
+From here, you could get very close to simply copy and paste the rest of a functional UI component into a LitElement for an early test of its capabilities. Leveraging scoped variables like this, rather than class properties, does leave you in much the same place that functional alternatives to UI development do; needing additional tools for memoization of that scope. As using LitElement means you're already in a class context, we can use the capabilities of a class directly rather than synthesizing them with memoization.
+
+A simpler approach to merging the two concepts is to set that fallback into the class property itself. Above I showed doing this in the `render()` lifecycle method, however, I find it much nicer to _only_ have the template surfaced therein. Leveraging one of the earlier lifecycle methods for managing defaults, validation, sanitation, transformation, or derivation help to maintain that structure. In this case, we'll use `willUpdate()` which doesn't require a `super` call or a returned value, but will always be visited during each render lifecycle:
+
+```js
+willUpdate(changedProperties: PropertyValues<this>): void {
+  this.closeDelay = this.closeDelay ?? 300;
+}
+```
+
+With this approach, you can most closely facilitate the single line defaulting that can be present when relying on a function to define a component. This does mean the `shouldRender()` the first method in the render lifecycle will not have your default value, and if that's an issue for your style of element development, you might want to move this fallback work there, but it also means that due to the fact the LitElement renders asynchronously there's technically a possibility that your local methods could be as well. Due to this fact, you may want to leverage a slightly more complete approach to a default.
+
+Here we see the property getter fallback to the value:
+
+```js
+get closeDelay(): number {
+  return this._closeDelay || 300;
+}
+set closeDelay(value: number) {
+  const closeDelay = value;
+  if (closeDelay === this.closeDelay) return;
+  this.requestUpdate('closeDelay', this.closeDelay);
+  this._closeDelay = closeDelay;
+}
+_closeDelay = 300;
+```
+
+This does eclipse the simplicity of a functional solution as we are choosing to apply the default by writing our own getter/setter pair on the property. At the same time, however, we've fallen into a possible trap of the functional fallback approach (and as the `willUpdate()` approach above), we're allowing our entire render lifecycle to be triggered for what might not be an actual change to our component state. Where `this.closeDelay` to already equal `300` and the application to change it to `undefined`, all of these approaches we've looked at so far would cause whatever could occur in your render lifecycle to occur without need.
```

Comment:
"Where `this.closeDelay` to already equal" -> "Were ..."
"lifecycle to occur without need" -> "lifecycle to occur needlessly"